Samples/BlazorExample/BlazorExample/Client/Pages/EditPerson.razor

```diff
@@ -33,17 +32,18 @@ else
         <td>Id</td>
         <td>@vm.Model.Id</td>
       </tr>
+      @if (vm.GetPropertyInfo<string>(nameof(vm.Model.Name)).CanRead)
```

Comment:
Have you considered using another component here in addition, a templated component?

We could combine a number of Blazor features to make the remaining markup very simple; templated components, a generic type parameter, and a cascading parameter.

The steps would be:

1) Create another component, let's say it's called EditRow. 
2) Into EditRow, copy this whole @if section
3) Replace <TextInput> with @ChildComponent, a parameter you define of type RenderFragment within EditRow. 
4) Add a generic @typeparam declaration for T
5) Add another parameter called For of type PropertyInfo<T>
6) Add a <CascadingValue> component to EditRow to cascade the value of @For into child components
7) Change TextInput to receive the PropertyInfo as a CascadingParameter rather than Parameter

If you did this, your markup for each property to be edited would be reduced to 3 lines:

<EditRow For="vm.GetPropertyInfo<string>(nameof(vm.Model.Name))">
    <TextInput>
</EditRow>

As well as simplifying the markup, I think this probably also reduces memory allocations, as you only call GetPropertyInfo() once, rather than 3 times.

Reply:
You'd mentioned this in gitter. And I do think it is an excellent idea - I just haven't figured out how to create such a component. Do you have an example or reference?

Reply:
Oh, I'm just an ideas man. Examples are implementation details! ;-)

Yep, I'm just updating Visual Studio, and then I'll see if I can knock something up and upload it to GitHub.

Reply:
Here you go. I think this does the job.

https://github.com/TheCakeMonster/Examples/tree/master/Blazor/TemplatedComponent/TemplatedCSLAEditRow